examples/pokemon-service-common/src/lib.rs

```diff
@@ -33,11 +37,16 @@ const PIKACHU_JAPANESE_FLAVOR_TEXT: &str =
     "ほっぺたの りょうがわに ちいさい でんきぶくろを もつ。ピンチのときに ほうでんする。";
 
 /// Rewrites the base URL of a request
-pub fn rewrite_base_url(base_url: String) -> impl Fn(Request) -> Request + Clone {
+pub fn rewrite_base_url(
+    scheme: Scheme,
+    authority: Authority,
+) -> impl Fn(Request) -> Request + Clone {
     move |mut req| {
         let http_req = req.http_mut();
-        let uri = format!("{base_url}{}", http_req.uri().path());
-        *http_req.uri_mut() = uri.parse().unwrap();
+        let mut uri_parts = http_req.uri().clone().into_parts();
+        uri_parts.authority = Some(authority.clone());
+        uri_parts.scheme = Some(scheme.clone());
+        *http_req.uri_mut() = Uri::from_parts(uri_parts).expect("valid uri");
```

Comment:
You're being misguided here and in the below usages by `.expect()`'s unfortunate naming into thinking that the message here is expressing what you expect, whereas this is supposed to be an error message :) See https://github.com/chris-morgan/rfcs/blob/unwrap_or_panic/text/0000-rename-expect.md#rationale-and-alternatives:

> [...] the name expect isn’t great on a couple of axes:
> It guides you in the direction of writing the message back to front, expressing what you expected to be the case, whereas it is supposed to be an error message when the expectation failed, meaning what went wrong, which is often expressed as the exact opposite.